test/legacy_test/test_initializer.py

```diff
@@ -811,6 +866,57 @@ def test_numpy_array_initializer_bf16(self):
         self.assertTrue(block.ops[1])
 
 
+class TestNumpyArrayInitializerPir(unittest.TestCase):
+    def setUp(self):
+        self.init_uniform_op_name = 'pd_op.uniform'
+        self.init_normal_op_name = 'pd_op.gaussian'
+        self.set_parameter_op_name = 'builtin.set_parameter'
+
+    def get_init_ops_by_op_name(self, block, op_name):
+        checked_ops = []
+        for op in block.ops:
+            # get init op
+            if op_name == op.name():
+                checked_ops.append(op)
+        return checked_ops
+
+    def test_numpy_array_initializer(self, dtype="float32"):
+        """Test the numpy array initializer with supplied arguments"""
+        import numpy
+
+        main = paddle.static.Program()
+        startup = paddle.static.Program()
+        np_array = numpy.random.random(10000).astype(dtype)
+        with paddle.static.program_guard(main, startup):
+            param = paddle.pir.core.create_parameter(
+                dtype=dtype,
+                shape=[5, 10],
+                name="param",
+                initializer=paddle.nn.initializer.Assign(np_array),
+            )
+
+            block = startup.global_block()
+            checked_ops = self.get_init_ops_by_op_name(
+                block, self.init_uniform_op_name
+            )
+            num_ops = 2 if dtype in ["float16", "uint16"] else 1
+            self.assertEqual(len(checked_ops), num_ops)
+            init_op = checked_ops[0]
+            self.assertEqual(init_op.type, 'assign_value')
+            assert (init_op.attr('fp32_values') == np_array).all()
```

Comment:
pir 模式下获取 attr 的方法和旧 ir 的不一样。可以先试一下下面的方法能不能运行通过
```suggestion
            assert (init_op.attrs()['fp32_values'] == np_array).all()
```